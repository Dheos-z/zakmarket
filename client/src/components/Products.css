.productsContainer {
    display: flex;
    width: 80%;
    margin: auto;
    justify-content: space-evenly;
    flex-flow: row wrap;
}

.flexItem {
    flex: 0 0 25em;
    width: 20%;
}

.flexItem + .flexItem {
    margin-top: 20px;
}

h2 {
    text-align: center;
}

.createProductDiv {
    text-align: center;
}
.createProductBtn {
    background-color: rgb(51, 99, 51);
    border: none;
  }