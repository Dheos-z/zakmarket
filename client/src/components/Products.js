import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import NavigationBar from './NavigationBar';
import Product from './Product';
import { CardDeck, CardColumns, CardGroup, Button } from 'react-bootstrap';
import './Products.css';


class Products extends Component {
    state = { products: [] };

    componentDidMount() {
        fetch(`http://localhost:3000/api/product`)
            .then(response => response.json())
            .then(json => this.setState({ products: json }));
    }

    handleDelete = idProductToDel => {
        // Delete from the json database using the API
        fetch('http://localhost:3000/api/product/' + idProductToDel, {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json'
            }
        }).then(response => response.json())
            .then(json => {
                alert(json.message || json.type);
            });

        // Delete from the state
        var { products } = this.state;
        for (var i = 0; i < products.length; i++) {
            if (products[i].idProduct === idProductToDel) {
                products.splice(i, 1);
                this.setState({ products });
            }
        }
    }


    render() {
        return (
            <div className="products">
                <NavigationBar />
                <br />
                <Link to='/createProduct'>
                    <div className="createProductDiv">
                        <Button className="createProductBtn">Create a new product</Button>
                    </div>
                </Link>
                <br />
                <h2>Our products</h2>
                <br />
                <div className="productsContainer">
                    <CardDeck>
                        {
                            this.state.products.map((product) => (
                                <Product
                                    key={product.idProduct}
                                    product={product}
                                    handleDelete={this.handleDelete}
                                />
                            ))
                        }
                    </CardDeck>
                </div>
            </div>
        );
    }
}

export default Products;