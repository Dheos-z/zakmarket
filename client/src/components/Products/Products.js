import React, { useState, useEffect } from 'react';
import { Link, BrowserRouter } from 'react-router-dom';
import NavigationBar from '../NavigationBar/NavigationBar';
import Product from '../Product';
import { Button } from 'react-bootstrap';
import './Products.css';
import history from '../../history';


const Products = () => {
    const [products, setProducts] = useState([]);

    useEffect(() => {
        fetch(`http://localhost:8080/api/product`)
            .then(response => response.json())
            .then(json => setProducts(json))
            .catch(error => {
                console.error(error);
            });
    });

    const handleDelete = idProductToDel => {
        // Delete from the json database using the API
        fetch('http://localhost:8080/api/product/' + idProductToDel, {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json'
            }
        }).then(response => response.json())
            .then(json => {
                console.log(json.message || json.type);
            })
            .catch(error => {
                console.error(error);
            });

        // Delete from the state
        for (var i = 0; i < products.length; i++) {
            if (products[i].idProduct === idProductToDel) {
                setProducts(products.splice(i, 1));
            }
        }
    }


    return (
        <div className="products">
            <NavigationBar />
            <br />
            <div className="createProductDiv">
                <BrowserRouter>
                    <Link to='/createProduct' onClick={() => history.push('/createProduct')}>
                        <Button className="createProductBtn">Create a new product</Button>
                    </Link>
                </BrowserRouter>
            </div>
            <br />
            <h2>Our products</h2>
            <br />
            <div className="productsContainer">
                {
                    products.map((product) => {
                        return (
                            <div className="flexItem" key={product.idProduct}>
                                <Product
                                    product={product}
                                    handleDelete={handleDelete}
                                />
                            </div>
                        )
                    })
                }
            </div>
        </div>
    );
}

export default Products;

