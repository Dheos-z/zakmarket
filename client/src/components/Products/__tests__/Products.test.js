import React from 'react';
import renderer from 'react-test-renderer';
import Products from '../Products';

describe('Products', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Products />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
