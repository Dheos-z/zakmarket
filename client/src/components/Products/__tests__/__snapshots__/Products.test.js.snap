// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products snapshot renders 1`] = `
<div
  className="products"
>
  <div
    className="navigationBar"
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Zak-Market
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/products"
            href="/products"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Products
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/createProduct"
            href="/createProduct"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Create a product
          </a>
        </div>
      </div>
    </nav>
  </div>
  <br />
  <div
    className="createProductDiv"
  >
    <a
      href="/createProduct"
      onClick={[Function]}
    >
      <button
        className="createProductBtn btn btn-primary"
        disabled={false}
        type="button"
      >
        Create a new product
      </button>
    </a>
  </div>
  <br />
  <h2>
    Our products
  </h2>
  <br />
  <div
    className="productsContainer"
  />
</div>
`;
